trinity/config.py

```diff
@@ -185,7 +191,10 @@ def vm_configuration(self) -> VMConfiguration:
         return self.genesis_data.vm_configuration
 
 
-class TrinityConfig:
+TSubConfig = TypeVar('TSubConfig', bound='BaseSubConfig')
+
+
+class BaseApplicationConfig:
```

Comment:
Ah! I see where my suggestion went sideways.

I was suggesting we **leave** the `TrinityConfig` name as is, and then call what is currently referred to as `SubConfig` an `ApplicationConfig` or `AppConfig`.  That lets the "trinity" application use the `TrinityConfig` and each other application that wants to run in trinity have their own `AppConfig`

Reply:
Gotcha. Changed the naming. Hope I got it right this time :)